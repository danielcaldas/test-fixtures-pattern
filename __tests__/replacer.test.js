const backgroundColorReplacer = require('../replacer');

describe('backgroundColorReplacer', () => {
    it('should replace color if they match', () => {
        const target = 'background-color: rgb(255, 0, 0)';
        const fromColor = [255, 0, 0];
        const toColor = [0, 255, 0];
        const options = { fromColor, toColor };

        expect(backgroundColorReplacer(target, options)).toMatchSnapshot();
    });

    it('should replace color if they match and add opacity when defined', () => {
        const target = 'background-color: rgb(255, 0, 0)';
        const fromColor = [255, 0, 0];
        const toColor = [0, 255, 0];
        const opacity = 0.5;
        const options = { fromColor, toColor, opacity };

        expect(backgroundColorReplacer(target, options)).toMatchSnapshot();
    });

    it('should not replace color if they do not match', () => {
        const target = 'background-color: rgb(254, 0, 0)';
        const fromColor = [255, 0, 0];
        const toColor = [0, 255, 0];
        const options = { fromColor, toColor };

        expect(backgroundColorReplacer(target, options)).toMatchSnapshot();
    });

    it('should not replace color there is no new color specified', () => {
        const target = 'background-color: rgb(253, 0, 0)';
        const fromColor = [255, 0, 0];
        const toColor = undefined;
        const options = { fromColor, toColor };

        expect(backgroundColorReplacer(target, options)).toMatchSnapshot();
    });
});
