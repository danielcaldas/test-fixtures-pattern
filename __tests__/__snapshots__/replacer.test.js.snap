// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backgroundColorReplacer should not replace color if they do not match 1`] = `"background-color: rgb(254, 0, 0)"`;

exports[`backgroundColorReplacer should not replace color there is no new color specified 1`] = `"background-color: rgb(253, 0, 0)"`;

exports[`backgroundColorReplacer should replace color if they match 1`] = `"background-color: rgb(0, 255, 0)"`;

exports[`backgroundColorReplacer should replace color if they match and add opacity when defined 1`] = `"background-color: rgba(0, 255, 0, 0.5)"`;
